README.md

```diff
@@ -159,6 +180,95 @@ ember exam --split=3 --partition=3
 
 **Note 2**: _You must be using Testem version `1.5.0` or greater for this feature to work properly._
 
+### Test Load Balancing
+
+```bash
+$ ember exam --load-balance=<num>
```

Comment:
@stefanpenner , let me help answer this one:

Your suggestion was initially how the arguments were designed 😄. From the #134 feedback we went with the proposed change that we believe simplifies the commands.

For context, lets look at how ember-exam can be used today:
1) `ember exam` —> same as `ember test`
2) `ember exam --split=6` —> spawns 1 browser, splitting the test suite into 6 partitions, run partition 1 only
3) `ember exam --split=6 --partition=1,2,3` —> spawns 1 browser, splitting the test suite into 6 partitions, run partition 1,2,3
4) `ember exam --split=6 --parallel` —> spawns 6 browsers, splitting the test suite into 6 partitions, run all 6 partitions
5) `ember exam --split=6 --partition=1,2,3 --parallel` —> spawns 3 browsers, splitting the test suite into 6 partitions, run partitions 1, 2, 3 in machine 1
6) `ember exam --split=6 --partition=4,5,6 --parallel` —> spawns 3 browsers, splitting the test suite into 6 partitions, run partitions 4, 5, 6 in machine 2

The number of browsers spawned is controlled by different parameters depending on the combination of commands used, which we thought was a bit convoluted.
eg. 2) and 3) without using —parallel, only 1 browser is spawned.
     4) if parallel is used without partitions, split determines the number of browsers.
     5) and 6) if partitions are passed, the number of partitions determines the number of browsers.


Proposed ember exam with test load balancing:
1’) `ember exam` —> basically run ember test
2’) `ember exam --load-balance=6` —> bring up 6 browser, sharing the full test suite with 6 browsers
3’) `ember exam --load-balance=6 --split=2 --partition=1` —> bring 6 browsers, splitting the test suite into 2 partitions, run partitions 1 in machine 1
4’) `ember exam --load-balance=6 --split=2 --partition=2` —> bring 6 browsers, splitting the test suite into 2 partitions, run partitions 2 in machine 1

Using `--load-balance=<number>` allows a single argument that set the number of browser to spawn.
This removes the need for `parallel`, because the load-balance value always define the number of browsers to run in parallel.
The use case of 2) to spawns a single browser to run one partition effectively gets replaced with `replay-execution`.
And 3) that spawns a single browser to run multiple partitions is not supported by `load-balance`.

Another important change is the use of `split` and `partition`, which predefines partitions of tests used in two ways in today’s ember-exam:
Firstly, it allows browsers to execute a unique subset of tests.
Secondly, this unique subset of tests can be defined to run across multiple CI machines.

With `load-balance`, predefined partitions of tests is replaced with dynamically load balanced of tests, this removes the need for `split` and `partitions` to split up the test suite in a static way *for each browser.*
The ability to run tests across CI machines is still preserved, by using `split` and `partition` to predefine partitions of tests for *a CI machine*.
Think of `split` as number of machines, and `partition` as identifier of the machine, as demonstrated by 3’) and 4’)

The run test across CI machine command gets a bit confusing if `--load-balance` is only a boolean:
A) `ember exam --load-balance=true --split=6 --partition=1,2,3 --parallel` —> bring 3 browsers, splitting the test suite into 6 partitions, load balance the tests from partitions 1, 2, 3 on machine 1
B) `ember exam --load-balance=true --split=6 --partition=4,5,6 --parallel` —> bring 3 browsers, splitting the test suite into 6 partitions, load balance the tests from partitions 4, 5, 6 on machine 2

Reply:
Thanks for the detailed explanation. The way I can reconcile these constraints, this is as follows:

*note:  just because what described bellow makes sense to me, doesn't mean it makes sense to a broader audience, I am sharing it to see if it resonates and as a basis for further discussion.*

* `--parallel` starts taking an optional argument, which represents the amount fo parallelism. This number can be `> 0` although higher numbers may be underutilized, with the wrong scheduling type.
* `--parallel` defaults to `0`, which simply mimics todays functionality
* `--load-balance` or `--share-work` or `--scheduler='stealing'` becomes a way to choose the mode of scheduling. This would ensure the work-stealing scheduler is opt-in.

Reply:
Thanks for the feedback @stefanpenner 

After some discussions offline, I feel your suggestions for the commands make sense going forward given:
- the possibility of making ember test adopt ember-exam's functionality
- load balancing become the default behavior, so the `--load-balance` command can go away

Taking your suggestion, the proposed commands will behave the same way as it does currently without `load-balance`. But with `load-balance`, it will introduce the new way to use `parallel` option to define the amount of parallelism.

Commands | Descriptions
-- | --
ember exam --split=6 | 1 browser, splitting the test suite into 6 partitions, run partition 1 only
ember exam --split=6 --partition=1,2,3 | 1 browser, splitting the test suite into 6 partitions, run partition 1,2,3
ember exam --split=6 --parallel | 6 browsers, splitting the test suite into 6 partitions, run all 6 partitions
ember exam --split=6 --partition=1,2,3 --parallel | 3 browsers, splitting the test suite into 6 partitions, run partitions 1, 2, 3 in machine 1
ember exam --split=6 --partition=4,5,6 --parallel | 3 browsers, splitting the test suite into 6 partitions, run partitions 4, 5, 6 in machine 2
  |  
ember exam --load-balance --parallel=6 | 6 browsers, sharing the full test suite with 6 browsers
ember exam --load-balance --split=2 --partition=1 --parallel=6 | 6 browsers, splitting the test suite into 2 partitions, run partitions 1 in machine 1
ember exam --load-balance --split=2 --partition=2 --parallel=6 | 6 browsers, splitting the test suite into 2 partitions, run partitions 2 in machine 1
  |  
ember exam --load-balance | 1 browsers, run full test suite using load-balance method
ember exam --load-balance --split=2 --partition=1 | 1 browser, splitting the test suite into 2 partitions, run partitions 1 in machine 1
ember exam --load-balance --split=2 --partition=2 | 1 browser, splitting the test suite into 2 partitions, run partitions 2 in machine 2


When `load-balance` becomes the default, the commands can be greatly simplified: (with some renaming of options like `split` to `partition` and `partition` to `run-part`)

Command | Descriptions
-- | --
ember exam | same as ember test
ember exam --filter='acceptance' | test with filter
ember exam --server | run test in server/dev mode
ember exam --random | run tests in random order
ember exam:iterate <num> |  
  |  
  | ** test distribution defaulted to load-balance
ember exam --parallel=6 | 6 browsers, sharing the full test suite with 6 browsers
ember exam --parallel=6 --partition=2 --run-part=1 | 6 browsers, splitting the test suite into 2 partitions, run part 1 in machine 1
ember exam --parallel=6 --partition=2 --run-part=2 | 6 browsers, splitting the test suite into 2 partitions, run part 2 in machine 1